crates/sui-framework/packages/sui-framework/sources/config.move

```diff
@@ -1,289 +1,282 @@
+            }),
+        };
+        field::add(&mut config.id, name, sobj);
+        option::none()
+    } else {
         let sobj: &mut Setting<Value> = field::borrow_mut(&mut config.id, name);
         let SettingData {
             newer_value_epoch,
             newer_value,
             older_value_opt,
         } = sobj.data.extract();
-        let (older_value_opt, removed_value) =
-            if (epoch > newer_value_epoch) {
-                // if the `newer_value` is for a previous epoch, move it to `older_value_opt`
-                (move newer_value, option::none())
-            } else {
-                // the current epoch cannot be less than the `newer_value_epoch`
-                assert!(epoch == newer_value_epoch);
-                (move older_value_opt, move newer_value)
-            };
-        let older_value_opt_is_none = older_value_opt.is_none();
-        sobj.data.fill(SettingData {
-            newer_value_epoch: epoch,
-            newer_value: option::none(),
-            older_value_opt,
-        });
-        if (older_value_opt_is_none) {
-            field::remove<_, Setting<Value>>(&mut config.id, name);
+        let (older_value_opt, removed_value) = if (epoch > newer_value_epoch) {
+            // if the `newer_value` is for a previous epoch, move it to `older_value_opt`
+            (move newer_value, move older_value_opt)
+        } else {
+            // the current epoch cannot be less than the `newer_value_epoch`
+            assert!(epoch == newer_value_epoch);
+            // if the `newer_value` is for the current epoch, then the option must be `none`
+            assert!(newer_value.is_none(), EAlreadySetForEpoch);
+            (move older_value_opt, option::none())
         };
+        sobj
+            .data
+            .fill(SettingData {
+                newer_value_epoch: epoch,
+                newer_value: option::some(value),
+                older_value_opt,
+            });
         removed_value
     }
+}
 
-    public(package) fun exists_with_type<
-        WriteCap,
-        Name: copy + drop + store,
-        Value: copy + drop + store,
-    >(
-        config: &Config<WriteCap>,
-        name: Name,
-    ): bool {
-        field::exists_with_type<_, Setting<Value>>(&config.id, name)
-    }
+#[allow(unused_mut_parameter)]
+public(package) fun remove_for_next_epoch<
+    WriteCap,
+    Name: copy + drop + store,
+    Value: copy + drop + store,
+>(
+    config: &mut Config<WriteCap>,
+    _cap: &mut WriteCap,
+    name: Name,
+    ctx: &mut TxContext,
+): Option<Value> {
+    let epoch = ctx.epoch();
+    if (!field::exists_(&config.id, name)) return option::none();
+    let sobj: &mut Setting<Value> = field::borrow_mut(&mut config.id, name);
+    let SettingData {
+        newer_value_epoch,
+        newer_value,
+        older_value_opt,
+    } = sobj.data.extract();
+    let (older_value_opt, removed_value) = if (epoch > newer_value_epoch) {
```

Comment:
This one not so sure about formatting. Even if `if` is on the same line, what to do with `else`? Or should both be newline + indent?

Reply:
We can discuss later, no need to block here for this PR.
I have another good, example to look at